src/effects.js

```diff
@@ -631,15 +633,25 @@ jQuery.each( {
 } );
 
 jQuery.timers = [];
+var oldLength = 0;
 jQuery.fx.tick = function() {
 	var timer,
 		i = 0,
 		timers = jQuery.timers;
 
 	fxNow = jQuery.now();
+	oldLength = timers.length;
+	for ( ; i < oldLength; i++ ) {
+		var removedTimers = 0;
 
-	for ( ; i < timers.length; i++ ) {
-		timer = timers[ i ];
+		for ( var j = 0; j < lis.length; j++ ) {
```

Comment:
hey @gibson042 ,

`jQuery.timers = [];
jQuery.fx.tick = function() {
	var timer,
		i = 0,
		timers = jQuery.timers;

	fxNow = jQuery.now();
	for ( ; i < timers.length; i++ ) {
		timer = timers[ i ];
                // storing the previous value of the 'timer'
		var x  = timers.indexOf(timer) 
		if (  !timer()  ) {
			if( x > timers.indexOf(timer)) {    
                              // checking and updating the value of  'i' ( if changed ).

				i = i - ( x - timers.indexOf(timer) ) 
			}
			if( timers[ i ] === timer ) {
				timers.splice( i--, 1 );
			}
		}
	}

	if ( !timers.length ) {
		jQuery.fx.stop();
	}
	fxNow = undefined;
};
`
This is the edited [ jQuery.fx.tick()] https://github.com/jquery/jquery/blob/master/src/effects.js#L644) function.
I would like to know if this fixes the first part of the bug before proceeding further.

Thank you ,
Preethi.